### DISCUSSION TITLE
How to check if field is required in typescript

### DISCUSSION THREAD

**Discussion by eddysims** [react-hook-form/react-hook-form#8215] [Q&A]
I have a React component written in typescript. It has a prop or `readonly rules?: RegisterOptions;`. I want to find out if the rules contain a `required` key to add the correct accessibility labels and what not. However, when I do this:

```ts
const isRequired = rules?.required?.value;
```

I get this:

![image](https://user-images.githubusercontent.com/46460840/163867375-047bbaaa-a245-4903-b829-965cccab2f29.png)

As far as I can tell, the `value` should be available through the `RegisterOptions` type.

![image](https://user-images.githubusercontent.com/46460840/163867601-7a6ff5a8-a51c-4d64-8778-2b52f218177b.png)

Am i doing something wrong here?

Link to the `validator` types:

https://github.com/react-hook-form/react-hook-form/blob/master/src/types/validator.ts

**Comment by Moshyfawn** [comment]
```ts
export type ValidationValue = boolean | number | string | RegExp;

export type ValidationRule<
  TValidationValue extends ValidationValue = ValidationValue,
> = TValidationValue | ValidationValueMessage<TValidationValue>;

export type ValidationValueMessage<
  TValidationValue extends ValidationValue = ValidationValue,
> = {
  value: TValidationValue;
  message: Message;
};
```

You gotta check if the value is an `object` to access `.value` type-safely
